### PR TITLE
Least common ancestor

### PR DESCRIPTION
**Pull request by estensland**
I added the start of complex query methods from PedHunter, a basic implementation of finding least common ancestors.

I am not sure it is the best way, but it is functional. If you think it should be done better, I can do some more research. It currently only returns the ancestor, but not the path from the individuals to the least common ancestors as discussed on PedHunter. Once the path is discovered, this can also return how two individuals are related..such as "grandchild", "second-cousin", "first cousin once removed", etc.

I you would rather wait on this request until I can create a system that returns not only the ancestor but also at least the path between the two people, I can do that no problem.

### CODE REVIEW

**Comment by masciugo** [comment]
Hi eric

It seems to me a good implementation but I have few things to mention before going to much further. 

First of all I wouldn't add the *path* feature, for the moment at least. Besides being too specific, the output of the method would be inconsistent with other query methods and couldn't be chained. We can think later, if necessary, a way to have it optional.

As you can see, your code is not Rails3 compatible. I use this gem in a Rails3 project but maybe it's time to remove this compatibility..

Lastly I'm going to add some little notes directly on your changes

File: README.md [hunk 1]
```diff
@@ -33,6 +33,9 @@ Some of them return a single AR object, like `father` and `paternal_grandfather`
 
 Some query methods can take options: `siblings(half: :mother)` returns maternal half-siblings (same mother). `paul.children(spouse: :titty)` returns all individuals that have paul as father and titty as mother.
 
+### Complex Query methods
+Inspired by the [PedHunter](http://www.ncbi.nlm.nih.gov/CBBresearch/Schaffer/pedhunter.html) tool, currently only the `least_common_ancestor`has been started. It returns an *ActiveRecord::Relation* of the first individuals to appear in the family tree of two individuals.
```
Comment (masciugo) [inline README.md]: maybe it's better to keep the name consistent with PedHunter method name as we refer to it calling the method *lowest_common_ancestors*. (plural as it, in general, returns *ActiveRecord::Relation*).

File: lib/genealogy/complex_query_methods.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        other_ancestor_record_ids += other_next_gen
+        other_parent_ids = other_next_gen
+
+        if (self_ancestor_record_ids & other_ancestor_record_ids).length > 0
+          return gclass.where(id: (self_ancestor_record_ids & other_ancestor_record_ids))
+        else
+          generation_count += 1
+        end
+      end
+      gclass.where(id: nil)
```
Comment (masciugo) [inline lib/genealogy/complex_query_methods.rb]: Only for rails 4 better use [none](http://api.rubyonrails.org/classes/ActiveRecord/QueryMethods.html#method-i-none) method:

`gclass.none`

**Comment by estensland** [comment]
All good points! Completely forgot about that Rails 3 issue. 

As for the 2+ and putting it on the class, I can change that. I like the thought of it being able to handle as much as possible. I can work on these changes no problem

**Pull request closed without merging**